db/migrate/20240405184242_create_orgs_reps_tables.rb

```diff
@@ -0,0 +1,146 @@
+      t.string :middle_initial
+      t.string :last_name
+      t.string :full_name
+      t.string :email
+      t.string :phone
+      t.string :address_type
+      t.string :address_line1
+      t.string :address_line2
+      t.string :address_line3
+      t.string :city
+      t.string :country_code_iso3
+      t.string :country_name
+      t.string :county_name
+      t.string :county_code
+      t.string :international_postal_code
+      t.string :province
+      t.string :state_code
+      t.string :zip_code
+      t.string :zip_suffix
+      t.jsonb :raw_address
+      t.float :lat
+      t.float :long
+      t.geography :location, limit: { srid: 4326, type: 'st_point', geographic: true }
+      t.timestamps
+    end
+    add_index :accredited_claims_agents, :full_name
+    add_index :accredited_claims_agents, :location, using: :gist
+
+    create_table :accredited_representatives, id: false, primary_key: :registration_number do |t|
+      t.string :registration_number, null: false
+      t.string :poa_code
+      t.string :first_name
+      t.string :middle_initial
+      t.string :last_name
+      t.string :full_name
+      t.string :email
+      t.string :phone
+      t.string :address_type
+      t.string :address_line1
+      t.string :address_line2
+      t.string :address_line3
+      t.string :city
+      t.string :country_code_iso3
+      t.string :country_name
+      t.string :county_name
+      t.string :county_code
+      t.string :international_postal_code
+      t.string :province
+      t.string :state_code
+      t.string :zip_code
+      t.string :zip_suffix
+      t.jsonb :raw_address
+      t.float :lat
+      t.float :long
+      t.geography :location, limit: { srid: 4326, type: 'st_point', geographic: true }
+      t.timestamps
+    end
+    add_index :accredited_representatives, :full_name
+    add_index :accredited_representatives, :location, using: :gist
+
+    create_table :accredited_organizations, id: false, primary_key: :poa_code do |t|
+      t.string :poa_code, limit: 3, null: false
+      t.string :name
+      t.string :phone
+      t.string :state, limit: 2
+      t.string :address_type
+      t.string :address_line1
+      t.string :address_line2
+      t.string :address_line3
+      t.string :city
+      t.string :country_code_iso3
+      t.string :country_name
+      t.string :county_name
+      t.string :county_code
+      t.string :international_postal_code
+      t.string :province
+      t.string :state_code
+      t.string :zip_code
+      t.string :zip_suffix
+      t.jsonb :raw_address
+      t.float :lat
+      t.float :long
+      t.geography :location, limit: { srid: 4326, type: 'st_point', geographic: true }
+      t.timestamps
+    end
+    add_index :accredited_organizations, :location, using: :gist
+    add_index :accredited_organizations, :name
+
+    create_join_table :accredited_organizations, :accredited_representatives do |t|
+      t.index :accredited_organization_poa_code
```

Comment:
I wonder if `t.references :accredited_organization, index: true, foreign_key: true` would do the same as this line + manually adding the foreign key?

Reply:
I update the migration to try that.